### PR TITLE
Remove list of known authorities, go back to one

### PR DESCRIPTION
**Pull request by MarkZuber**

### CODE REVIEW

File: src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs [hunk 1]
```diff
@@ -610,13 +593,12 @@ public T AddKnownAdfsAuthority(string authorityUri, bool isDefaultAuthority)
         /// </summary>
         /// <param name="authorityUri">Azure AD B2C authority, including the B2C policy (for instance
         /// <c>"https://fabrikamb2c.b2clogin.com/tfp/{Tenant}/{policy}</c></param>)
-        /// <param name="isDefaultAuthority">Boolean telling if this is the default authority
-        /// for the application</param>
+        /// <param name="validateAuthority">TODO(migration): documentation</param>
         /// <remarks>You can add several authorities, but only one can be the default authority.</remarks>
         /// <returns>The builder to chain the .With methods</returns>
-        public T AddKnownB2CAuthority(string authorityUri, bool isDefaultAuthority)
+        public T WithB2CAuthority(string authorityUri, bool validateAuthority)
```
Comment (jennyf19) [inline src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs]: just a thought....b2c users should always pass validateAuthority=false. 
can we have b2c devs pass in only the authorityUri, and we set validateAuthority to false for them? just one less thing for them to think about and less documentation for b2c when using msal.

Comment (MarkZuber) [reply src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs]: Yup, will do.

Comment (jennyf19) [reply src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs]: Then a todo will be to let b2c know so they can update their docs. :) I'll add this to the onenote somewhere

File: src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs [hunk 2]
```diff
@@ -593,12 +593,11 @@ public T WithAdfsAuthority(string authorityUri, bool validateAuthority)
         /// </summary>
         /// <param name="authorityUri">Azure AD B2C authority, including the B2C policy (for instance
         /// <c>"https://fabrikamb2c.b2clogin.com/tfp/{Tenant}/{policy}</c></param>)
-        /// <param name="validateAuthority">TODO(migration): documentation</param>
         /// <remarks>You can add several authorities, but only one can be the default authority.</remarks>
         /// <returns>The builder to chain the .With methods</returns>
-        public T WithB2CAuthority(string authorityUri, bool validateAuthority)
+        public T WithB2CAuthority(string authorityUri)
```
Comment (jennyf19) [inline src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs]: awesome

File: src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs [hunk 3]
```diff
@@ -593,14 +577,13 @@ private static string GetAadAuthorityAudienceValue(AadAuthorityAudience authorit
...
         /// </summary>
         /// <param name="authorityUri">Authority URL for an ADFS server</param>
-        /// <param name="isDefaultAuthority">Boolean telling if this is the default authority
-        /// for the application</param>
+        /// <param name="validateAuthority">TODO(migration): documentation</param>
         /// <remarks>You can add several authorities, but only one can be the default authority.
         /// MSAL.NET will only support ADFS 2019 or later.</remarks>
         /// <returns>The builder to chain the .With methods</returns>
-        public T AddKnownAdfsAuthority(string authorityUri, bool isDefaultAuthority)
+        public T WithAdfsAuthority(string authorityUri, bool validateAuthority)
```
Comment (henrik-me) [inline src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs]: >validateAuthority [](start = 61, length = 17)

Can we have a default/override where this is true? both for aad and adfs?

Comment (MarkZuber) [reply src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs]: I've moved aad/adfs methods to be true for default of validateAuthority.

**Pull request merged**